app/models/pinned_article.rb

```diff
@@ -0,0 +1,32 @@
+class PinnedArticle
+  class << self
+    def exists?
+      Settings::General.feed_pinned_article_id.present?
+    end
+
+    def id
+      Settings::General.feed_pinned_article_id
+    end
+
+    def updated_at
+      return if Settings::General.feed_pinned_article_id.blank?
+
+      setting = Settings::General.find_by(var: :feed_pinned_article_id)
+      setting.updated_at
+    end
+
+    def get
+      return if Settings::General.feed_pinned_article_id.blank?
+
+      Article.published.find_by(id: Settings::General.feed_pinned_article_id)
```

Comment:
Good point. If the article no longer exists, we should nullify `Settings::General.feed_pinned_article_id` as well :) I'll see if it's convenient to add an exception or not. I don't want the exception to get to the browser, the admin should be able to smoothly repin anything else if the article is gone

Reply:
Actually, now that I think about it. It would behave the same way as it does if the article is put in draft in the meantime. It will just "disappear" from the feed. I'll double check to make sure

Reply:
Ok I checked. I can't pin new posts if the article is either put to draft or deleted. I'll address this